diff -ur icu/source/icudefs.mk.in icu.new/source/icudefs.mk.in
--- icu/source/icudefs.mk.in	2010-08-30 20:59:02.000000000 +0200
+++ icu.new/source/icudefs.mk.in	2011-03-27 15:41:18.000000000 +0200
@@ -150,7 +150,7 @@
 ifeq ($(ENABLE_RPATH),YES)
 RPATHLDFLAGS = $(LD_RPATH)$(LD_RPATH_PRE)$(libdir)
 endif
-LDFLAGS = @LDFLAGS@ $(RPATHLDFLAGS)
+LDFLAGS = @LDFLAGS@ -Wl,-rpath -Wl,$(RPATHLDFLAGS)
 
 # What kind of libraries are we building and linking against?
 ENABLE_STATIC = @ENABLE_STATIC@
